util/circular-chk.py

```diff
@@ -0,0 +1,51 @@
+import glob
+import re
+from sys import exit, argv
+from pathlib import Path
+
+strict = False
+if len(argv) > 1 and argv[1] == '--strict' :
+    strict = True
+
+regexExclude = r"keymaps\/(.*)via\/"
+reg = re.compile(regexExclude)
+filesToscan = Path('keyboards').glob('**/rules.mk')
+illegalSRC = list()
+
+for f in filesToscan:
+    p = str(f).replace('\\','/')
+    basepath = p.replace('keyboards/','')
+    basepath_arr = basepath.split('/')
+
+    # Skip 'via' file
+    if reg.search(p):
+        continue
+
+    # Read file as lines
+    lineList = list()
+    with f.open() as file:
+        try:
+            lineList = file.readlines()
+        except:
+            # sometimes encoding issues with japanese characters for example
+            pass
+
+    # Concat multilines
+    lineList = '\n'.join(lineList).replace('\\\n','').split('\n')
+    for l in lineList:
+        if not l.startswith('SRC ='):
+            continue
+        imports = l.replace('SRC =','').replace('\t','').split(' ')
+        for i in imports:
+            _i = i.split('/')
+            if(_i[0] == 'keyboards' and _i[1] != basepath_arr[0]):
+                illegalSRC.append(f)
```

Comment:
Can you explain the intent behind this code? I don't understand why it checks only the top-level subfolder, which is often a vendor folder that doesn't even have a `rules.mk`.

Reply:
problem was some people imported funcs from other vendors, so had to be sure the imports were in the current tree.

Reply:
I think perhaps the criteria are either not well defined or not well explained. What if there's a circular import between `clueboard/rev1` and `clueboard/rev2`, wouldn't this miss that?

Reply:
It was mean to fix `illegal` imports not `circular`. Is circular a problem? because those keyboards would build if there is circular import anyway no?

Reply:
I'm going by what's written in the title and description. What is an illegal import then?

Reply:
Legal:
file: keyboards/zeal60/rules.mk
content:
```
SRC =	keyboards/zeal60/zeal60.c \
		keyboards/zeal60/rgb_backlight.c \
```
Illlegal:
file: keyboards/ergoFoobar/rules.mk
content:
```
SRC =	keyboards/zeal60/zeal60.c \
		keyboards/zeal60/rgb_backlight.c \
```

Reply:
What makes those illegal? Can we find a set of terminology that is descriptive and doesn't require examples to explain?

Reply:
importing a file that does not belong to your tree in the keyboards tree. I don't know what terminology we can use for this.

Reply:
I feel like we need a bunch of use cases to test mentally what this is expected to catch and validate that it works as expected

Reply:
As needs may evolved it would be better to open an issue in `cli` write down cases and migrate it to cli.